rcl/test/rcl/test_events.cpp

```diff
@@ -220,19 +257,95 @@ wait_for_msgs_and_events(
   return subscription_event_ready || publisher_event_ready;
 }
 
+TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_unsupported_lifespan) {
+  if (is_unsupported) {
+    rmw_time_t deadline {0, 0};
+    rmw_time_t lifespan {1, 0};
+    rmw_time_t lease_duration {1, 0};
+    rmw_qos_liveliness_policy_t liveliness_policy = RMW_QOS_POLICY_LIVELINESS_AUTOMATIC;
+    EXPECT_EQ(RMW_RET_ERROR,
```

Comment:
I would expect all of these to be receiving back `RMW_RET_UNSUPPORTED` if they're failing because it is unsupported. Are we masking that error code at some layer? 

https://github.com/aws-robotics/rmw/blob/qos_reviewed/rmw/include/rmw/ret_types.h#L31

Reply:
That will not be possible, since the init function does not return RMW_RET_T, but instead a pointer: nullptr if unable to initialize, or a filled ptr.

Reply:
We could always change the function prototype :dagger:

Reply:
Yeah, we can add an issue to the backlog to add a return error code from this function. It's certainly not a blocker for this PR.